From 5c672e25de9ef89e942ea83940e2c82abd03b8cc Mon Sep 17 00:00:00 2001
From: yuanzheng <tsinghuayuan86@gmail.com>
Date: Thu, 18 Nov 2010 14:33:27 +0800
Subject: [PATCH 1/1] Add some macro judgment for ubuntu-10.10.

Add some macro judgment for ubuntu-10.10.
Because the file "bits/poll.h" of ubuntut-10.10 add a judgment
about whether the macro "__USE_XOPEN2K8" is defined, but this
judgment is not existing in other ubuntu version. So we need
add some macro judgment to avoid it.

Signed-off-by: yuanzheng <tsinghuayuan86@gmail.com>
---
 utils/uart_console/uart_console.c |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/utils/uart_console/uart_console.c b/utils/uart_console/uart_console.c
index ec92830..5249990 100644
--- a/utils/uart_console/uart_console.c
+++ b/utils/uart_console/uart_console.c
@@ -53,14 +53,20 @@
 #include <sys/systeminfo.h>
 #include <sys/stream.h>
 #include <thread.h>
-#else
+#else //no define __svr4__
+#if !defined POLLRDNORM
 const uint32_t POLLRDNORM = 0x040; 
+#endif
+#if !defined POLLRDBAND
 const uint32_t POLLRDBAND = 0x080; 
-//const uint32_t POLLPRI = 0x002; 
-//const uint32_t POLLOUT = 0x004; 
+#endif
+#if !defined POLLWRNORM
 const uint32_t POLLWRNORM = 0x100; 
+#endif
+#if !defined POLLWRBAND
 const uint32_t POLLWRBAND = 0x200;
 #endif
+#endif //__svr4__
 
 static int setup_term();
 static int setup_netlink(char * hostp, char * portp);
-- 
1.6.3.3

